signer/tests/integration/transaction_coordinator.rs

```diff
@@ -123,8 +139,169 @@ async fn get_signer_public_keys_and_aggregate_key_falls_back() {
     let rotate_keys_signer_set: BTreeSet<PublicKey> =
         rotate_keys.signer_set.iter().copied().collect();
 
-    assert_eq!(rotate_keys.aggregate_key, aggregate_key);
+    assert_eq!(rotate_keys.aggregate_key, aggregate_key.unwrap());
     assert_eq!(rotate_keys_signer_set, signer_set);
 
     testing::storage::drop_db(db).await;
 }
+
+/// Test that we run DKG if the coordinator notices that DKG has not been
+/// run yet.
+///
+/// This test proceeds by doing the following:
+/// 1. Create a database, an associated context, and a Keypair for each of
+///    the signers in the signing set.
+/// 2. Populate each database with the same data, so that they have the
+///    same view of the canonical bitcoin blockchain. This ensures that
+///    they participate in DKG.
+/// 3. Check that there are no DKG shares in the database.
+/// 4. Start the [`TxCoordinatorEventLoop`] and [`TxSignerEventLoop`]
+///    processes for each signer.
+/// 5. Once they are all running, signal that DKG should be run. We signal
+///    them all because we do not know which one is the coordinator.
+/// 6. Check that we have exactly one row in the `dkg_shares` table.
+/// 7. Check that they all have the same aggregate key in the `dkg_shares`
+///    table.
+///
+/// Some of the preconditions for this test to run successfully includes
```

Comment:
You could circumvent this by manually grabbing the `Settings::new_from_default_config()` as `mut` and setting the bootstrap set to known values, then passing clones into the `TestContext`'s builder's `with_settings()`.

Reply:
Ahh, good to know. 🙏🏽